client/protocols/export.py

```diff
@@ -0,0 +1,124 @@
+from client.protocols.common import models
+from client.utils import auth
+from client.utils import network
+from datetime import datetime
+from urllib import error
+import logging
+import os
+import pickle
+
+log = logging.getLogger(__name__)
+
+GAE_DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
+
+EXPORT_CACHE = "export_cache.pkl"
+
+class ExportProtocol(models.Protocol):
+    """Downloads submissions from the server"""
+
+    def run(self, messages):
+        """If the --export parameter was used, downloads all submissions 
+        for the current assignment from the server and then exits
+        """
+        if not self.args.export:
+            return
+        
+        self.access_token = auth.authenticate(self.args.authenticate)
+        log.info('Authenticated with access token %s', self.access_token)
+        
+        data = None
+        try:
+            data = pickle.load(open(EXPORT_CACHE, "rb"))
+        except IOError:
+            data = {
+                'endpoint': self.assignment.endpoint   
+            }
+            data['assign'], data['course'] = self.get_ids(data['endpoint'])
+            data['students'] = self.get_students(data['course'])
+            if not data['students']:
+                return
+            data['current'] = 0
+        
+        current_student = 0
+        try:
+            if not os.path.exists('submissions'): 
```

Comment:
"submissions" should be stored in a global variable at the top of the file (e.g. `SUBMISSION_DIR = 'submissions'`). You also use 'submissions' on line 82.